compiler/semdata.nim

```diff
@@ -238,8 +249,11 @@ proc makeVarType*(c: PContext, baseType: PType): PType =
     addSonSkipIntLit(result, baseType.assertNotNil)
 
 proc makeTypeDesc*(c: PContext, typ: PType): PType =
-  result = newTypeS(tyTypeDesc, c)
-  result.addSonSkipIntLit(typ.assertNotNil)
+  if typ.kind == tyTypeDesc:
```

Comment:
Last time I thought about it, typedesc[typedesc[T]]  is not the same as typedesc[T] and conflating these is not sound.